docs/command-reference.md

```diff
@@ -1,6 +1,6 @@
 # Command reference
 
-- [Workload](command-reference/tanzu-apps-workload.md)
+- [Workload](command-reference/tanzu_apps_workload.md)
```

Comment:
make a separate PR for doc name changes

Reply:
I changed this here because I put some docs in the commands reference folder and when I ran `make docs`to add the new flags, all of them were erased. So I tried to correct at least these two here.